aten/src/ATen/native/cuda/Nonzero.cu

```diff
@@ -37,9 +37,12 @@ __global__ void write_indices(
     int64_t* inp,
     TensorDims<index_t> dims,
     int ndim,
-    index_t n) {
+    index_t n,
+    int64_t * total = nullptr,
+    int64_t fill_value = -1) {
   auto index = threadIdx.x + blockIdx.x * blockDim.x;
-  if (index < n) {
+  bool cond = (total == nullptr || index < (uint)*total);
+  if (index < n && cond) {
```

Comment:
This will compare a 32bit wide (`index`) with `index_t` width `n`, is this intentional (we can assume the number of threads along `x`, `index < 2**31)`?

Reply:
Good catch, we need to compute `index` in int64_t